controller/machine/controller.go

```diff
@@ -109,6 +111,14 @@ func (m *Lifecycle) Remove(obj *v3.Machine) (*v3.Machine, error) {
 	if obj.Status.MachineTemplateSpec == nil {
 		return obj, nil
 	}
+	singleEtcdOrControl, err := checkSingleNode(obj, m.clusters, m.machineClient)
+	if err != nil {
+		return obj, err
```

Comment:
should return nil for *v3.Machine when no changes happened in the object.

Reply:
Actually we should return error, so the delete will be retried. User can fix the cluster by adding more machines to it, and the initial machine removal should be retried.